### PR TITLE
fix(desktop,gui-app,shared): make the local host's lifecycle target-independent and self-healing

### PR DESCRIPTION
**Pull request by hdkshingala**
A developer on a MacBook whose app was bound to a remote host (their iMac) found Settings ▸ Host ▸ Overview offering a **Start host** button for a "Not installed" local row, next to a window card saying the host was unavailable. Two rulings came out of that report; this PR implements both.

## 1. The local lifecycle is independent of which host is active

The selection authority's ensure was gated *"only when the local host is wanted"* (the redesign's C5): a user with a healthy preferred remote never paid for an idle local boot. That rule is reversed — `requestLocalEnsureIfWanted` → **`requestLocalEnsureIfDown`**, so a dead or never-dialed local host is ensured whichever host a window is pointed at, paced by the same 30 s cooldown and retry hold.

Only the **narration** stays target-scoped: the window narrator still speaks for the window's own host, so a background local ensure while a remote serves is silent.

Two companion rules were needed for that to be correct:

| Rule | Why |
| --- | --- |
| `applyDamping` bypasses the return-to-target window for an incumbent held usable **only** by the engine's own in-flight ensure (`isHeldOnlyByOwnEnsure`) | Otherwise "window failed over onto L, P came back, L dies" hands the 20 s window a `connecting` host that serves nobody. A test caught this. |
| the desktop ensure port maps `busy` → `{ok: true}` | A host with work in progress **is** running. Otherwise a never-dialed non-target local host with a pending byte swap draws a CLI converge every 30 s forever — nothing dials a non-target host, so nothing else can end never-dialed. |

## 2. A failed boot retries on its own, without a re-sign-in

`armFirstInstallOnSignIn` → **`armLocalHostBootOnSignIn`**, with both halves of its contract widened:

- **Owed while no host is RUNNING** (`activation === "unavailable"`), not while no bytes are installed. A half-completed first install — bytes landed, service never started — is still owed a boot instead of reading as done on the first status read.
- **Retries on `LOCAL_HOST_BOOT_RETRY_LADDER_MS`** (30 s → 60 s → 2 min → 5 min cap) after any outcome that did not leave a host running: `deferred` (CLI lock held by another Traycer process), `failed`, `installed-not-converged`, a fingerprint mismatch, or a throw. Previously the only thing that tried again was the next sign-in **edge**, which for a user who stays signed in never comes.
- Settles on `ok`/`busy`, on a status read showing a live runtime, or on the removal sentinel.

Consent is unchanged and still read at the moment of acting: signed out means nothing happens until an account signs in; a sign-out cancels a pending retry and resets the ladder; a host the user removed is never reinstalled. A same-state `signedIn: true` change (a token refresh) does not bypass the ladder's pacing. Rung 0 matches the authority's `LOCAL_ENSURE_RETRY_COOLDOWN_MS`, and where both actors find an installed host down at launch the controller coalesces the two `ensure:false` intents onto one job.

## 3. No Start verb

Overview's down-local cluster loses **Start host** — it was a second process actor for a host the app already brings back, and under `removedByUser` it was a silent no-op (`convergeReady` short-circuits under the sentinel). What remains:

- **Run doctor**, disabled while this machine's lifecycle lane is busy.
- **Reinstall Traycer**, only in the one state the automation deliberately leaves alone: it clears the sentinel, then converges (new `useRunnerReinstallTraycer`).

The sidebar's *"Viewing — this window runs on X"* named the app-wide **active host**, not the physical machine; it now reads *"the active host is X"*.

## Verification

- `bun run compile` green (5 projects); eslint + prettier clean on every changed file.
- `clients/shared` full suite **76 files / 1273 tests**; selection-authority engine **101/101** (A6 inverted for the target-independent ensure, a new pin for the incumbent rule with both halves, the corpse test made async so the cold-boot ensure settles).
- desktop selection-ports **33/33**; desktop `host-launch-converge` **43/43** — the boot-actor block reworked with fake timers to prove the retry lands *without* a sign-in edge, at the right rung boundaries, that a half-install is still owed, `busy`/`deferred` handling, that a sign-out cancels and resets the ladder, and that the disposer clears a pending timer.
- gui-app settings/host suites 13 files / 109, plus a new 4-case `host-overview-local-down.test.tsx`.

## Known residual

A deliberate `traycer host stop` of a **non-active** local host mid-session is not fought: nothing dials it, so no evidence reaches the engine, and the health monitor treats pid.json-gone as a deliberate stop — same as released builds. The doctor sheet's own fix action for a `host-start` issue is still labelled "Start host" (a repair inside a diagnostic flow), left as is.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: clients/desktop/src/electron-main/startup/desktop-startup.ts [hunk 1]
```diff
@@ -952,11 +952,12 @@ export function runDeferred<
...
   // Two DIFFERENT actions, deliberately not merged. The reconciler settles the
-  // debt of a host that exists; the first install creates one that never has,
-  // and only for a signed-in user (see `armFirstInstallOnSignIn`). Arming is
+  // debt of a host that exists, once; the boot actor gets a host RUNNING -
+  // installing one that never existed if need be - only for a signed-in user,
+  // and keeps retrying until it is (see `armLocalHostBootOnSignIn`). Arming is
   // synchronous and cheap - it either acts now or waits for the sign-in that
   // the pre-retirement renderer gate also waited for.
-  armFirstInstallOnSignIn(services.hostController, services.signedIn);
+  armLocalHostBootOnSignIn(services.hostController, services.signedIn);
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/desktop/src/electron-main/startup/desktop-startup.ts]: Fixed in 189c18b2. Real finding — the disposer was discarded, and it matters more now than before, since the arm owns a retry ladder as well as the subscription.

`runDeferred` now captures it and registers it on the bridge teardown list:

```ts
const disposeLocalHostBoot = armLocalHostBootOnSignIn(
  services.hostController,
  services.signedIn,
);
state.bridge?.disposeFns.push(disposeLocalHostBoot);
```

`TState` gains exactly the constraint that registration needs (`{ readonly bridge: { readonly disposeFns: Array<() => void> } | null }`) and nothing more, so the generic still accepts a light test state; `BootState` satisfies it structurally, and a null bridge keeps todays behaviour. Validated: `bun run compile` exit 0, `clients/desktop` startup suites 10 files / 89 tests pass.

File: clients/gui-app/src/components/settings/panels/__tests__/host-overview-local-down.test.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    const management = buildOverviewManagement({
+      getRemovalState: vi.fn(() => Promise.resolve({ removedByUser: false })),
+    });
+    renderLocalDown({ settingUp: true, management, name: "This Mac" });
+
+    const doctorButton = await screen.findByTestId(
+      "host-overview-recovery-doctor",
+    );
+    expect(doctorButton.hasAttribute("disabled")).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/components/settings/panels/__tests__/host-overview-local-down.test.tsx]: Fixed in 189c18b2 — the guideline applies and this is a new file, so all four lookups in the suite now use role queries with the accessible name and the native property:

```ts
const doctorButton = await screen.findByRole<HTMLButtonElement>("button", {
  name: "Run doctor",
});
expect(doctorButton.disabled).toBe(false);
```

Worth recording why this is safe rather than merely tidier: the status card renders a second "Run doctor" (`host-overview-status-card.tsx`), but it lives inside a `DropdownMenuContent` that is unmounted while closed, so the accessible name stays unambiguous in this scenario. The suite passes 4/4 against the real render, which is what proves it. The remaining `queryByTestId` calls are absence assertions ("no Start verb", "no Reinstall") — a role query by name cannot express "this testid is gone" as precisely, so those stay.

File: clients/shared/host-selection/__tests__/selection-authority-engine.test.ts [hunk 3]
```diff
@@ -3296,18 +3315,106 @@ describe("SelectionAuthorityEngineImpl - P1.3 failover scenarios", () => {
...
+    authority2.engine.ingestEvidence(
+      "A",
+      incarnation2,
+      dialOutcome("P", "revive", "success", clock.now()),
+    );
+    expect(authority2.engine.snapshot().effectiveHostId).toBe("L");
+    clock.advance(RETURN_TO_TARGET_STABILITY_MS);
+    expect(authority2.engine.snapshot().effectiveHostId).toBe("P");
+
+    authority2.dispose();
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/shared/host-selection/__tests__/selection-authority-engine.test.ts:3425]: Adjusted in 189c18b2, though not for the stated reason — the premise is inaccurate and I want the record to be right: both authorities were already disposed. Line 3417 was `authority2.dispose()` and line 3418 immediately below it was `authority.dispose()`, so nothing leaked past the test.

The substantive half of the point does stand: disposing the first engine AFTER `clock.advance(RETURN_TO_TARGET_STABILITY_MS)` means it still takes timer callbacks from an advance that exists to drive the control. It could not change a verdict (no assertion above that point is revisited, and the two engines share no state but the clock), but it is not a fixture worth reasoning about, so `authority.dispose()` now runs before the control is built. Suite passes 101/101.

File: clients/desktop/src/electron-main/startup/host-launch-converge.ts [hunk 4]
```diff
@@ -179,29 +208,70 @@ export async function refreshHostRegistryIfNotRemoved(
...
+
+  const clearRetry = (): void => {
+    if (retryTimer !== null) {
+      clearTimeout(retryTimer);
+      retryTimer = null;
+    }
+  };
 
   const dispose = (): void => {
+    clearRetry();
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/desktop/src/electron-main/startup/host-launch-converge.ts:250]: Fixed in 84ed7831. Confirmed real, and worth saying that the previous commit is what made it REACHABLE: registering the disposer on the bridge teardown list (`189c18b2`, from a sibling thread) means a quit landing mid-ladder is precisely this race. `dispose()` now records terminal before releasing anything:

```ts
const dispose = (): void => {
  // TERMINAL, and that has to be recorded BEFORE the resources go, because
  // an attempt can be in flight right now. [...]
  settled = true;
  clearRetry();
  ...
};
```

In-flight work is not cancellable (no token on the IPC round trip), so the continuation still runs — it just cannot re-arm.

Pinned by a new test, `the disposer fences an attempt that is already in flight`, which holds the `convergeReady` resolver so disposal lands provably inside the in-flight window, then resolves `failed` and advances past the whole ladder. Mutation-probed: dropping the `settled = true` line fails that test and only that test (43 others still pass), then restored byte-identically.

File: clients/gui-app/src/hooks/runner/use-runner-reinstall-traycer-mutation.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    mutationFn: async () => {
+      if (hostManagement === null) {
+        throw new Error("Host provisioning is not available on this platform.");
+      }
+      await hostManagement.clearRemoval();
+      const outcome = await hostManagement.convergeReady(false);
+      if (outcome.kind === "ok" || outcome.kind === "busy") {
+        return outcome;
+      }
+      throw new Error(outcome.message);
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/hooks/runner/use-runner-reinstall-traycer-mutation.ts:76]: Fixed in 84ed7831 — a real dead end, and your reading of the cause is exactly right: `clearRemoval` has already persisted, `onSettled` refetches false, the verb disappears, and main is not going to cover it because its boot actor settled at launch on the sentinel it saw then.

The cluster now keeps the verb while the last attempt is in error:

```ts
// The verb SURVIVES ITS OWN FAILURE. [...]
const removalRepairable = removed || reinstall.isError;
```

On the alternative you offered — handing the cleared state to a retrying lifecycle actor — I deliberately did not go there. Removal is the ONE state where automation is off by consent, and the click is the consent; re-arming main from a renderer-initiated sentinel clear means new IPC and a second actor in the one place the design says there should be none. The backstop is real rather than theoretical: the next launch reads a cleared sentinel, sees `activation: "unavailable"`, and boots on the retry ladder with no user action. Within the session, the button is the retry.

Known limit, stated rather than papered over: `isError` is mutation state, so navigating away and back drops the affordance until relaunch.

Pinned by a new test, `a reinstall whose converge fails keeps the verb, because the sentinel is already cleared` — the sentinel returns true once and false thereafter, and the assertion waits for the refetch to have actually happened before checking the button, so it cannot pass on a query that never re-ran. Mutation-probed: reverting to `{removed ? ...}` fails that test and only that test.

File: clients/shared/host-selection/selection-authority-engine.ts [hunk 6]
```diff
@@ -2178,10 +2213,7 @@ export class SelectionAuthorityEngineImpl implements SelectionAuthorityEngine {
       return false;
     }
 
-    const targetHostId = this.preferredHostId ?? localHostId;
-    if (targetHostId !== localHostId && this.isUsable(targetHostId, leases)) {
-      return false;
-    }
+    // Deliberately NO "is the target serving?" gate here - see the doc above.
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/shared/host-selection/selection-authority-engine.ts:2216]: Fixed in 4bdeef1b. Confirmed real — I traced all four links before touching anything: `host-controller.ts` returns `{kind:"ok", value:{running:false, version:null}}` under the sentinel, the port collapsed every `ok`, and `resolveLocalEnsure` calls `onHostProvedAlive(token.hostId)` on `ok`, which clears the refusal streak and drops `usableSince` so the lease reads usable. Your last link — failover then selecting it, with each 30s no-op ensure re-clearing the streak the failed dials build — follows from those three.

The port now preserves it:

```ts
if (outcome.kind === "ok") {
  if (outcome.value.running) return { ok: true };
  return { ok: false, reason: "removed-by-user", deferred: false };
}
```

I took your second option (preserve `running:false`) rather than the first (gate the ensure for removed hosts): the engine has no business knowing about a desktop consent sentinel, and this port is exactly the adapter that knows what the controller means. Deliberately NOT `deferred: true` — deferred means "nothing ran, judge nothing", but here we know something definite, that the host is not there. A plain failure lets the cooldown pace it, and each retry costs one sentinel read until the user reinstalls, at which point the very next ensure converges for real.

Worth noting the collapse predates this PR — what this PR changed is the population it applies to, from "the local host is the window target" to "always", which is what turns it into a live failover hazard. Pinned by a new test, `refuses to call a REMOVED host alive, even though its converge answers ok`, mutation-probed: collapsing the branch back to `{ok: true}` fails it and only it (33 siblings pass), then restored byte-identically.

File: clients/gui-app/src/components/settings/panels/host-overview-panel.tsx [hunk 7]
```diff
@@ -984,59 +989,89 @@ export function HostUpdateRequiredSlot(props: {
...
+  const removed = removal.data?.removedByUser === true;
+  // The verb SURVIVES ITS OWN FAILURE. `reinstall` clears the removal
+  // sentinel first and converges second, so a converge that comes back
+  // `failed`/`deferred` leaves the sentinel already cleared - `removed` flips
+  // to false, and this cluster would drop the only affordance for a machine
+  // that still has no host. Nothing else picks it up in this session either:
+  // main's boot actor settled at launch on the sentinel it saw then. Keeping
+  // the button on `isError` makes the retry the user's to take now; the next
+  // launch sees a cleared sentinel and boots on the ladder without them.
+  const removalRepairable = removed || reinstall.isError;
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/components/settings/panels/host-overview-panel.tsx]: Fixed in 03657e26 — real, and a regression in the fix from two commits ago rather than an edge case of it. Confirmed the mechanism: the second click flips the mutation error -> pending, `isError` goes false, and the sentinel is already cleared so `removed` is false too; both terms gone, the button unmounts for the whole attempt and takes the inline spinner with it.

```ts
const removalRepairable = removed || reinstall.isError || reinstall.isPending;
```

`busy` already carries `reinstall.isPending`, so it renders disabled with its progress rather than merely present.

The part worth recording: my earlier test could not have caught this, because on the FIRST attempt `removed` is still true until the sentinel refetch lands — the button stays mounted for the wrong reason. The new test, `a second reinstall attempt keeps the button mounted while it is in flight`, waits for that refetch (asserting the removal read reached a second call) and only then clicks again, parking the second converge on a released gate so the assertion happens while it is provably still in flight. Mutation-probed: dropping the `isPending` term fails that test and only it.

File: clients/gui-app/src/hooks/runner/use-runner-reinstall-traycer-mutation.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      // converge that followed it failed, and a removal-state read that still
+      // says "removed" would keep offering Reinstall for a host that is now
+      // merely not installed.
+      if (hostManagement !== null) {
+        void queryClient.invalidateQueries({
+          queryKey: runnerQueryKeys.hostRemovalState(hostManagement),
+        });
+      }
+    },
+    onSuccess: () => {
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/hooks/runner/use-runner-reinstall-traycer-mutation.ts]: Fixed in 03657e26. Confirmed against the outcome contract: `installed-not-converged` is documented as the bytes having committed with only the post-commit service/readiness invariant failing, and `failed` can follow a service cycle that already wrote its record — so gating those two invalidations on success meant Settings kept reporting "Not installed" for a machine whose install had landed. All three reads now invalidate from `onSettled`.

On how it is pinned — this one could not be tested through the panel, and I want to be explicit rather than quietly test something weaker. The only query mounted in that tree on `hostInstalledRecord` is `useEmptyAccountLocalRecoveryZone`, whose `enabled` requires `scope.host === null`, which is the exact opposite of the down-local state the Reinstall verb renders in; `traycerCli` is `undefined` in those fixtures besides. So a UI-level assertion on either key would have passed no matter what this hook did. That was measured, not assumed: an instrumented run had `installedRecord` at 0 calls while the sentinel read reached 2 in the same render. The test is therefore at the hook level, spying on `invalidateQueries` and asserting the exact instance-keyed entries, with the sentinel key as a positive control so "nothing ran at all" cannot pass it.

One thing I found and deliberately did NOT change: the sibling `useRunnerConvergeReady` has the same `onSuccess`-only shape, so the identical window exists on the ordinary converge path. It is pre-existing and outside this PR, so it is not in this diff — flagging it here rather than widening the changeset.

File: clients/desktop/src/electron-main/startup/host-launch-converge.ts [hunk 9]
```diff
@@ -179,29 +208,81 @@ export async function refreshHostRegistryIfNotRemoved(
...
   const dispose = (): void => {
+    // TERMINAL, and that has to be recorded BEFORE the resources go, because
+    // an attempt can be in flight right now. Clearing the timer and the
+    // subscription alone leaves `settled` false, so the continuation of an
+    // `getStatus()`/`convergeReady()` that resolves after disposal walks into
+    // `scheduleRetry()` and arms a fresh timer - a disposed actor that goes on
+    // provisioning. Reachable in production since the disposer is registered
+    // on the bridge's teardown list: a quit that lands mid-ladder is exactly
+    // this race.
+    settled = true;
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/desktop/src/electron-main/startup/host-launch-converge.ts:249]: Fixed in 053886c8. Correct, and a fair catch on an incomplete fix rather than a new bug: recording terminal in `dispose()` covers the RE-ARM, and I stopped there. It leaves the in-flight continuation free to walk past the removal, activation and consent checks into `convergeReady(false)` — a CLI `host ensure` starting against a controller the app is tearing down.

The guard is now read again after the await, immediately:

```ts
const status = await hostController.getStatus();
// DISPOSAL RE-READ, and it belongs here for the same reason the consent
// re-read below does: the decision to act was taken before a round trip,
// and teardown can land inside it. [...]
if (settled) return;
```

Which makes it the third thing this attempt re-reads after its round trip, alongside `removedByUser` and `isSignedIn` — the shape was already there, disposal was just missing from it.

Pinned by `the disposer stops an attempt BEFORE it can start provisioning`: `getStatus` parks on a promise the test releases, `dispose()` runs while the attempt is provably inside that await, and the account stays SIGNED IN so nothing else in the continuation could turn it back — without the guard, the status it then sees (never installed, `activation: "unavailable"`) sends it straight into `convergeReady`. Mutation-probed: removing the line fails that test and only that test (44 siblings pass), then restored byte-identically. Desktop startup suites 91/91, `bun run compile` exit 0.

File: clients/desktop/src/electron-main/selection/desktop-selection-ports.ts [hunk 10]
```diff
@@ -682,26 +682,59 @@ export class DesktopLocalHostOutageSignal implements LocalHostOutageSignal {
...
+      // `ok`, which is what lets proof of life create the evidence record that
+      // ends never-dialed. It used to resolve `deferred` (paced retry, lease
+      // left alone), which was harmless while the ensure only fired for a
+      // WANTED local host: a window pointed at it dialed it and ended
+      // never-dialed on its own. The ensure now fires whichever host is
+      // effective, and a local host nothing dials would otherwise draw one
+      // converge - a CLI spawn - every pacing hold for as long as it stayed
+      // busy. The pending swap is not this port's business: the launch
+      // reconciler and the idle-apply monitors own updates.
+      if (outcome.kind === "busy") return { ok: true };
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/desktop/src/electron-main/selection/desktop-selection-ports.ts]: Fixed in 40ca107f. You are right, and the CLI says so in its own words — I went and read `assertHostNotBusy` rather than reasoning from the name:

> Throws `E_HOST_BUSY` when a LIVE host has work in progress, **or when a live host's idle/busy state can't be determined (fail-safe)** […] Any reachable-but-unprobeable outcome (404 from a pre-feature host, malformed body, connect error, or timeout) is treated as busy (fail-safe)

So `busy` means the PID is alive and nothing more. A wedged host is exactly the indeterminate case, and `{ok: true}` handed it to `onHostProvedAlive`: refusal evidence cleared, lease usable, failover free to choose it, and each later ensure clearing the refusals its failed dials had just rebuilt.

`busy` rejoins `deferred` — lease untouched, request paced. `failed` keeps its own answer, since it ran and concluded, so it stays the one arm that may arm the dead-lease cooldown.

Worth being explicit about what I got wrong, because it is the same mistake twice on this branch. I introduced that mapping HERE, to spare a non-target local host one CLI spawn per pacing hold: nothing dials such a host, so nothing else ends never-dialed. That is a cost argument, and I let it buy a correctness property. The spawn comes back, and it is bounded and self-limiting — real work finishes, the next converge answers `ok` with a version it proved. Restoring the shortcut would need reachability independently confirmed, and this port has no such evidence; the engine's own dial is what has it. This is the second finding of the same shape as the removed-host one, both mine, both from treating an `ok`-ish outcome as an answer to a question it was not asked.

The test is inverted rather than added, and mutation-probed: restoring `{ok: true}` fails it and only it (33 siblings pass), then restored byte-identically. Full desktop suite 1563/1563, `bun run compile` exit 0.

File: clients/gui-app/src/hooks/runner/use-runner-reinstall-traycer-mutation.ts [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+  return useMutation<MutationOutcome<ConvergeReadyOk>>({
+    mutationKey: runnerMutationKeys.reinstallTraycer(),
+    mutationFn: async () => {
+      if (hostManagement === null) {
+        throw new Error("Host provisioning is not available on this platform.");
+      }
+      await hostManagement.clearRemoval();
+      const outcome = await hostManagement.convergeReady(false);
+      if (outcome.kind === "ok" || outcome.kind === "busy") {
+        return outcome;
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/hooks/runner/use-runner-reinstall-traycer-mutation.ts]: Fixed in 5dacf2b1, and you spotted the same defect I had just fixed one layer down — the ensure-port mapping — sitting unfixed in the hook I wrote. `busy` here is the CLI declining to swap a live host's bytes, so nothing was reinstalled, and as a fail-safe it is not even evidence the host serves.

While fixing it I found the sibling case one line away and fixed that too: `ok` with `running: false` is `convergeReady`'s removal-sentinel short-circuit, which AFTER `clearRemoval` means the clear did not take — the converge did nothing, under consent that was supposed to have been revoked, and it would have toasted success and dropped the verb just the same. Leaving it would have repeated the pattern that has already cost this branch three rounds: fix the case the reviewer named, ship the one beside it.

```ts
if (outcome.kind === "ok" && outcome.value.running) return outcome;
if (outcome.kind === "ok") {
  throw new Error("Traycer was reinstalled, but no host started on this computer. Run doctor to see why.");
}
throw new Error(outcome.message);
```

Both reject, so `onError` says why and `isError` keeps the verb on the page. This is now a DELIBERATE divergence from `useRunnerConvergeReady`, which still resolves `busy`, and the doc comment says so: that hook asks "is the host up?", where a busy answer is informative; this one asks "did the reinstall happen?", where it is not.

Two new hook-level tests, both mutation-probed: restoring the old `ok || busy` condition fails both and nothing else. gui-app hooks + settings panels 1230/1230, `bun run compile` exit 0.

File: clients/desktop/src/electron-main/startup/host-launch-converge.ts [hunk 12]
```diff
@@ -179,121 +208,211 @@ export async function refreshHostRegistryIfNotRemoved(
...
-          //
-          // Left ARMED deliberately, matching the throw arm: the next sign-in
-          // edge attempts again. `inFlight` clears in `finally`, so nothing
-          // blocks that retry.
-          log.warn("[host-controller] first install did not complete", {
+        if (outcome.kind === "ok" || outcome.kind === "busy") {
+          // `busy` is the CLI's HOST_BUSY: a live host with active work
+          // declined a byte swap. That is a running host, which is all this
+          // actor wants - the same reading the authority's ensure port makes.
+          settle();
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/desktop/src/electron-main/startup/host-launch-converge.ts]: Fixed in 3f887471 — the third site of this same mistake on the branch, and the one that mattered most, because settling is permanent for the process. `busy` now earns the next rung like every other resolved non-ok.

Thank you for naming the concrete reason nothing downstream covers it. I had been reasoning "the authority ensures a down local host anyway", and that is only true for a host the fleet can NAME: `readLastKnownLocalHostId` returns null outright for an UNUSABLE enrollment record — it deliberately does not fall back to `pid.json` there — so `requestLocalEnsureIfDown` returns on its second line and the wedge waits for a relaunch. That is exactly the "no automatic recovery without user action" this branch exists to remove, so settling on `busy` was working against its own purpose.

`ok` still settles. Its `running: false` variant can only arrive under the removal sentinel, where settling IS the right answer, so that one stays.

The test is inverted rather than added, and it asserts both halves: the arm stays subscribed AND the rung actually fires. A listener-count assertion alone would have passed against an arm that kept its subscription and never acted. Mutation-probed: restoring `|| busy` fails it and only it (44 siblings pass), then restored byte-identically. Full desktop suite 1563/1563, `bun run compile` exit 0.

For the record, since this is now three: the mapping was wrong in the ensure port (r3811136124), in the reinstall hook (r3811229216), and here. All three were me reading an `ok`-shaped outcome as an answer to a question it was not asked. I have swept the remaining `busy` handling in this diff — `runLaunchHostConvergeReconcile`'s apply arm treats `busy` as a pass-through, which is correct there: it is the controller's own gate declining to chase a mutation with a recovery, not a claim about reachability.

**Comment by hdkshingala** [comment]
@coderabbitai approve
